### PR TITLE
Fix #139 - "Can't find license-maven-plugin"

### LINKED ISSUE #139: "Can't find license-maven-plugin" in 1.16 if plugin isn't configured in the POM
If you try to run `mvn license:aggregate-add-third-party` without adding the plugin to the root / parent POM first, you get the error `Can't find license-maven-plugin` throws in this line:

https://github.com/mojohaus/license-maven-plugin/blob/70b96dfd6db8b5bde085567ea3cb2ef1bdef2d83/src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java#L197

I see two solutions:

1. Enhance the error message: "Please add the plugin to your POM"
2. Put the three values in a property file (using resource filtering) and load that file from the classpath.

### PR DESCRIPTION
**Pull request by AndyGee**
Use the PluginDescriptor.
Most painful checkstyle.

### CODE REVIEW

File: src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java [hunk 1]
```diff
@@ -203,11 +204,30 @@ protected void doAction()
         }
         if ( groupId == null )
         {
-            throw new IllegalStateException( "Can't find license-maven-plugin" );
+            try
+            {
+                final PluginDescriptor pd = ( PluginDescriptor ) getPluginContext().get( "pluginDescriptor" );
+                groupId = pd.getGroupId();
+                artifactId = pd.getArtifactId();
+                version = pd.getVersion();
```
Comment (ppalaga) [inline src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java]: Isn't this actually the preferred way to get the `g:a:v` of the current plugin? If so, I guess we can remove the attempts above (startig at https://github.com/mojohaus/license-maven-plugin/pull/348/files#diff-38b03aa7a6c189ea0f7e814313a14682R182 ) to get it via ` project.getPluginArtifactMap()` and `project.getPluginManagement().getPluginsAsMap().get( "org.codehaus.mojo:license-maven-plugin" )` ?

**Comment by ppalaga** [comment]
OK, now I see there is 0374cf4 that delivers the test coverage.

File: src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java (hunk 1 above)
Comment (AndyGee) [reply src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java]: On premise, never touch a running system. I just didn't want to throw that away. I also just updated an existing IT by removing the build dep. I don't think the fix needs it's own IT.

**Comment by AndyGee** [comment]
Hmm, not that simple it seems, without digging in, is there a step thad does magic on the invoker.properties?

File: src/it/add-third-party-no-encoding/invoker.properties [hunk 2]
```diff
@@ -19,5 +19,5 @@
 # <http://www.gnu.org/licenses/lgpl-3.0.html>.
 # #L%
 ###
-invoker.goals=clean license:add-third-party
+invoker.goals=clean org.codehaus.mojo:license-maven-plugin:@pom.version@:add-third-party
```
Comment (ppalaga) [inline src/it/add-third-party-no-encoding/invoker.properties]: I think you want `${project.groupId}:${project.artifactId}:${project.version}:add-third-party` see https://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html#invokerPropertiesFile

**Comment by AndyGee** [comment]
That does it - no more build dep required for a command line - I'd not remove the legacy code in this request, but I'd agree it should be removed through new request/issue.

**Comment by AndyGee** [comment]
The CI jobs are actually passing, but need an update to go green.

**Comment by ppalaga** [comment]
I do not see the CI passing. I restarted Appveyor on Java 8 and I am investigating how to fix the broken Java installation in TravisCI.

**Comment by ppalaga** [comment]
The master is now passing the whole OS x Java version matrix. Could you please rebase and also squash into a single commit?

**Comment by AndyGee** [comment]
The CI is flaky

**Comment by ppalaga** [comment]
Yeah, the timeouts on Appveyor do not look good.

**Comment by ppalaga** [comment]
I cleaned the caches on TravisCI and restarted. Let's see if it helps.

**Comment by ppalaga** [comment]
Maven Central timeouts seems to be the cause of the random failures on the CI. AppVeyor passed for this PR, TravisCI passed here: https://github.com/mojohaus/license-maven-plugin/pull/350

**Pull request merged**